main/main.go

```diff
@@ -0,0 +1,259 @@
+	}
+
+	if err := utils.WriteToBucket(bucketName, paths.cveBucketSubPath, paths.cveFileNameWithPath); err != nil {
+		return err
+	}
+	log.Infof("done pushing to bucket path %q", filepath.Join(bucketName, paths.cveBucketSubPath))
+
+	if err := utils.WriteToBucket(bucketName, paths.cveHashBucketSubPath, paths.cveHashFileNameWithPath); err != nil {
+		return err
+	}
+	log.Infof("done pushing to bucket path %q", filepath.Join(bucketName, paths.cveHashBucketSubPath))
+
+	return nil
+}
+
+func getPaths(path string, c component) (fileAndBucketPaths, error) {
+	switch c {
+	case k8s:
+		return fileAndBucketPaths{
+			cveFileNameWithPath:     filepath.Join(path, k8sCveFileName),
+			cveHashFileNameWithPath: filepath.Join(path, k8sCveHashFileName),
+			cveBucketSubPath:        filepath.Join(os.Getenv(utils.GCloudBucketPrefix), kubernetesBucketDir, k8sCveFileName),
+			cveHashBucketSubPath:    filepath.Join(os.Getenv(utils.GCloudBucketPrefix), kubernetesBucketDir, k8sCveHashFileName),
+		}, nil
+
+	case istio:
+		return fileAndBucketPaths{
+			cveFileNameWithPath:     filepath.Join(path, istioCveFileName),
+			cveHashFileNameWithPath: filepath.Join(path, istioCveHashFileName),
+			cveBucketSubPath:        filepath.Join(os.Getenv(utils.GCloudBucketPrefix), istioBucketDir, istioCveFileName),
+			cveHashBucketSubPath:    filepath.Join(os.Getenv(utils.GCloudBucketPrefix), istioBucketDir, istioCveHashFileName),
+		}, nil
+
+	default:
+		return fileAndBucketPaths{}, fmt.Errorf("unknown component type %d ", c)
+	}
+}
+
+func getK8sAndIstioCVEs(feedReader map[string]string) ([]utils.NvdCVEEntry, []utils.NvdCVEEntry, error) {
+	var allK8sCVEs, allIstioCVEs []utils.NvdCVEEntry
+
+	for feedName, fileName := range feedReader {
+		dat, err := ioutil.ReadFile(fileName)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "error reading file %q", fileName)
+		}
+
+		cves, err := utils.GetCVEs(dat)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "error unmarshalling file %q", fileName)
+		}
+
+		k8sCVEs := filterCVEs(cves, "kubernetes")
+		log.Infof("found %d k8s cves for year %s", len(k8sCVEs), feedName)
+		allK8sCVEs = append(allK8sCVEs, k8sCVEs...)
+
+		istioCVEs := filterCVEs(cves, "istio")
+		log.Infof("found %d istio cves for year %s", len(istioCVEs), feedName)
+		allIstioCVEs = append(allIstioCVEs, istioCVEs...)
+	}
+	log.Infof("total %d k8s and %d istio CVEs found", len(allK8sCVEs), len(allIstioCVEs))
+	return allK8sCVEs, allIstioCVEs, nil
+}
+
+func filterCVEs(cves *utils.NvdCVEs, project string) []utils.NvdCVEEntry {
+	var cveEntries []utils.NvdCVEEntry
+	for _, cve := range cves.Entries {
+		if appliesToProject(cve, project) {
+			cveEntries = append(cveEntries, cve)
+		}
+	}
+	return cveEntries
+}
+
+func appliesToProject(cve utils.NvdCVEEntry, project string) bool {
+	for _, node := range cve.Configurations.Nodes {
+		for _, cpeMatch := range node.CPEMatch {
+			if strings.Contains(cpeMatch.CPE23Uri, fmt.Sprintf("%s:%s", project, project)) {
```

Comment:
Q for @connorgorman : Are vendor and package always specified? Or do we need to check for, e.g., `cpe23:kubernetes:*:...` as well?

Reply:
In any case, consider store the `fmt.Sprintf` expression in a variable computed once at the beginning of the function